pipeline/thumbnails.go

```diff
@@ -0,0 +1,93 @@
+package pipeline
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/livepeer/catalyst-api/clients"
+	"github.com/livepeer/go-tools/drivers"
+)
+
+const framesEverySecs = 5
+
+func GenerateThumbs(input *url.URL, output *url.URL) error {
+	in, err := clients.SignURL(input)
+	if err != nil {
+		return err
+	}
+	tempDir, err := os.MkdirTemp(os.TempDir(), "thumbs-*")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tempDir)
+
+	args := []string{
+		"-i", in,
+		"-vf", "select=eq(pict_type\\,I)",
+		"-vsync", "vfr",
+		"-vf", fmt.Sprintf("fps=1/%d", framesEverySecs),
+		path.Join(tempDir, "/keyframes_%03d.jpg"),
+	}
```

Comment:
I think it would be cleaner to move these ffmpeg into it's own place like `video/thumbs.go` similar to transmux/clip/etc?

Reply:
Also i think this incantation of ffmpeg is simple enough for us to use go-ffmpeg instead of cmd.Run right?

Reply:
Also these args could use some comments to explain what they're doing.

Reply:
Ah yep all good points will do 👍

Reply:
Ah.. just remembered there are two `vf` options so can't use the map in go-ffmpeg

Reply:
> I think it would be cleaner to move these ffmpeg into it's own place like `video/thumbs.go` similar to transmux/clip/etc?

Definitely agree it should be separated, for now I've created a new package just for thumbnails though to keep it simple. Then I'll do a follow up PR to rearrange a couple of things as it seems like we might have too much going on in the video package.